engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/CreateProcessInstanceProcessor.java

```diff
@@ -199,4 +214,95 @@ private DeployedProcess getProcess(final long key, final CommandControl controll
 
     return process;
   }
+
+  private void activateElementsForStartInstructions(
+      final ProcessInstanceCreationRecord record,
+      final DeployedProcess process,
+      final long processInstanceKey,
+      final ProcessInstanceRecord processInstance) {
+    stateWriter.appendFollowUpEvent(
+        processInstanceKey, ProcessInstanceIntent.ELEMENT_ACTIVATING, processInstance);
+    stateWriter.appendFollowUpEvent(
+        processInstanceKey, ProcessInstanceIntent.ELEMENT_ACTIVATED, processInstance);
+
+    final Map<DirectBuffer, Long> activatedFlowScopeIds = new HashMap<>();
+    activatedFlowScopeIds.put(processInstance.getElementIdBuffer(), processInstanceKey);
+    record
+        .startInstructions()
+        .forEach(
+            instruction -> {
+              final DirectBuffer elementId =
+                  new UnsafeBuffer(StringUtil.getBytes(instruction.getElementId()));
+              final long flowScopeKey =
+                  activateFlowScopes(process, processInstanceKey, elementId, activatedFlowScopeIds);
+
+              final long elementKey = keyGenerator.nextKey();
+              final ProcessInstanceRecord elementRecord =
+                  createProcessInstanceRecord(process, processInstanceKey, elementId, flowScopeKey);
+              commandWriter.appendFollowUpCommand(
```

Comment:
:x: I think this could activate an element twice. E.g. given this process:
![image](https://user-images.githubusercontent.com/14032870/172837320-d0eec1c8-15d7-4b8d-b0e5-10275564077e.png)

With  `client.createProcessInstance().startBefore("task").startBefore("subprocess").send().join()` the sub process would be activated twice. The first time as part of `activateFlowScopes()` for `task` and the second time when the second start instruction is processed.

Reply:
I think you are right. It's a bit of an odd scenario. What would the expected behaviour be in this case? 
Send a command to activate the subprocess, and then send a command to activate the task. This would sort of create 2 process instances in one 😅 Then the task would be activated twice.

Reply:
It is valid user input though. 

Could be a misunderstanding. Maybe the customer didn't know we will activate the surrounding event scopes and so they want to do it explicitly

Reply:
> With client.createProcessInstance().startBefore("task").startBefore("subprocess").send().join() the sub process would be activated twice. The first time as part of activateFlowScopes() for task and the second time when the second start instruction is processed.

Yes, it would create the subprocess twice. For me, this is the expected behavior. 

The rule for process instance modification is: it creates a new element instance in the existing scope instance. If no scope instance exists yet then a new scope instance is created.

The start-process-anywhere command follows the same rule. 

Independent of the order of the two instructions, the scope instance of the subprocess doesn't exist yet. As a result, it will create a new scope instance of the subprocess twice.

I'll create a new test case for it.